kmath-histograms/src/commonMain/kotlin/kscience/kmath/histogram/Counters.kt

```diff
@@ -1,20 +1,49 @@
 package kscience.kmath.histogram
 
-/*
+import kotlinx.atomicfu.atomic
+import kotlinx.atomicfu.getAndUpdate
+import kscience.kmath.operations.RealField
+import kscience.kmath.operations.Space
+
+/**
  * Common representation for atomic counters
- * TODO replace with atomics
  */
+public interface Counter<T : Any> {
+    public fun add(delta: T)
+    public val value: T
+    public companion object{
+        public fun real(): ObjectCounter<Double> = ObjectCounter(RealField)
+    }
+}
+
+public class IntCounter : Counter<Int> {
```

Comment:
Why not inline class?

Reply:
It is an interface. Atomics should not be exposed to the outside world. Also, inlining does not give anything here.